packages/hub/src/graphql/types/Searchable.ts

```diff
@@ -0,0 +1,107 @@
+import {
+  groupRoute,
+  modelRoute,
+  relativeValuesRoute,
+  userRoute,
+} from "@/routes";
+import { builder } from "../builder";
+import { Group } from "./Group";
+import { Model } from "./Model";
+import { RelativeValuesDefinition } from "./RelativeValuesDefinition";
+import { User } from "./User";
+
+type Tag = "Model" | "RelativeValuesDefinition" | "User" | "Group";
+
+function tagged(obj: any, tag: Tag) {
+  obj._searchableTag = tag;
+  return obj;
+}
+
+const SearchableObject = builder.unionType("SearchableObject", {
+  types: [Model, RelativeValuesDefinition, User, Group],
+  resolveType: (wrappedObj) => {
+    const tag: Tag = (wrappedObj as any)._searchableTag;
+    switch (tag) {
+      case "Model":
+        return Model;
+      case "RelativeValuesDefinition":
+        return RelativeValuesDefinition;
+      case "User":
+        return User;
+      case "Group":
+        return Group;
+      default:
+        throw new Error("Incorrectly tagged object");
+    }
+  },
+});
+
+export const Searchable = builder.prismaNode("Searchable", {
```

Comment:
does this search models that are private?

Reply:
Yes; see https://github.com/quantified-uncertainty/squiggle/pull/2674#discussion_r1421072670